src/parser/expression.rs

```diff
@@ -200,7 +207,7 @@ fn parse_infix(input: ParserInput, left: Expression) -> InternalParserResult<Exp
                 Operator::Slash => InfixOperator::Slash,
                 Operator::Star => InfixOperator::Star,
             };
-            let precedence = get_precedence(remainder);
```

Comment:
This was the crux of the issue. At this point `remainder` is the input just _after_ the operator, so `get_precedence` wasn't actually matching against the right token and  always fell through to the default case, which is `Precedence::Lowest`.

The fix instead gets precedence using the operator token directly.